models/agent_model.go

```diff
@@ -631,6 +647,32 @@ func (s Agent) TemplateDelimiters(svc *Service) *DelimiterPair {
 	return &tdp
 }
 
+// HashPassword func to calculate password hash
+var HashPassword = func(password, salt string) string {
+	buf, err := bcrypt.GenerateFromPasswordAndSalt([]byte(password), bcrypt.DefaultCost, []byte(salt))
+	if err != nil {
+		log.Fatal(err, "cannot calculate hash for password")
```

Comment:
let's return wrapped error instead of calling `log.Fatal`

Reply:
but we still need somewhere to `panic` of `fatal`?

Reply:
we should log and return error message and retry again. For one failing resource we shouldn't stop all of them.